src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.Analysis.Tree.cs

```diff
@@ -599,13 +601,22 @@ private static Scope CreateNestedScope(BoundNode node, Scope parentScope, Closur
                     return newScope;
                 }
 
-                private void DeclareLocals<TSymbol>(Scope scope, ImmutableArray<TSymbol> locals)
+                private void DeclareLocals<TSymbol>(Scope scope, ImmutableArray<TSymbol> locals, bool declareAsFree = false)
                     where TSymbol : Symbol
                 {
                     foreach (var local in locals)
                     {
                         Debug.Assert(!_localToScope.ContainsKey(local));
-                        _localToScope[local] = scope;
+                        if (declareAsFree)
+                        {
+#if DEBUG
+                            _freeVariables.Add(local);
+#endif
+                        }
+                        else
+                        {
+                            _localToScope[local] = scope;
```

Comment:
>_localToScope[local] = scope; [](start = 28, length = 29)

Should we use Add method instead to ensure that we are doing this only once per local?